pkg/datastore/example_noinstancer_test.go

```diff
@@ -30,7 +29,10 @@ func ExampleDataStore_noInstancer() {
 	p2 := &xUser{uId, "John", 36, "Prod"}
 	p3 := &xUser{"P3", "Pat", 39, "Dev"}
 
```

Comment:
This example (`ExampleDataStore_noInstancer`) is showing an incorrect usage of this library - an entity (`xUser`) that supports the concept of multiple instances is registered with DAL, but the instancer was not passed to DAL. Do we really need to show the developers using this library examples how not to use the library? \
When I first see this:
![Screen Shot 2023-04-06 at 2 24 49 PM](https://user-images.githubusercontent.com/2506593/230496916-f031c5e9-35e4-493f-8282-c3dc8c631ecb.png)
The first thing that comes to my mind is that the first example shows how DAL deals with an entity that supports the notion of multiple instances, while the 2nd example shows how DAL deals with an entity that does not support that. But that's not the case.

Reply:
@jeyhun  Can you expand on why you think it's not the case? The second example is without Instancer which means there is no access to data that is specific to the instance.

Reply:
@jeyhun Fixed the comments in noinstance test